cranelift/codegen/src/remove_constant_phis.rs

```diff
@@ -412,7 +412,7 @@ pub fn do_remove_constant_phis(func: &mut Function, domtree: &mut DominatorTree)
                     new_actuals.push(actual_i, &mut func.dfg.value_lists);
                 }
             }
-            func.dfg[edge.inst].put_value_list(new_actuals);
+            *func.dfg.insts[edge.inst].value_list_mut().unwrap() = new_actuals;
```

Comment:
Pre-existing: I think this function leaks the portion of the pool corresponding to `old_actuals`. Doesn't need to be fixed in this PR, I just wanted to note it.

Reply:
Should we open an issue?

Reply:
https://github.com/bytecodealliance/wasmtime/issues/5451